app/src/main/java/com/pwhs/quickmem/presentation/app/search/SearchViewModel.kt

```
@@ -19,27 +29,75 @@ class SearchViewModel @Inject constructor() : ViewModel() {
+                _uiEvent.send(SearchUiEvent.ShowError("Search query not found."))
+            }
+        }
+    }
+
+    private fun deleteAllSearchHistory() {
+        viewModelScope.launch {
+            try {
+                searchQueryRepository.clearSearchHistory()
+                loadSearchHistory()
+                _uiEvent.trySend(SearchUiEvent.ClearAllSearchResent)
+            } catch (e: Exception) {
```

detekt

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)